spec/Model.spec.ts

```diff
@@ -943,8 +943,15 @@ describe('Model', () => {
         description: 'Example description'
       })
 
+      const note2 = store.add('notes', {
+        id: '11',
+        description: 'Example description'
+      })
+
       todo.notes.add(note)
       note.description = 'something different'
+      todo.notes.replace([note, note2])
```

Comment:
this takes a snapshot, which was making `dirtyAttributes` compare against the latest version of the model instead of the original one.